src/Lunr/Core/Configuration.php

```diff
@@ -149,6 +149,29 @@ public function load_file(string $identifier): void
         $this->size_invalid = TRUE;
     }
 
+    /**
+     * Load environment variables as configuration.
+     *
+     * @return void
+     */
+    public function load_environment(): void
```

Comment:
This should take a parameter, defining the prefix of the environment variables to load. Like, `LUNR` loading all environment variables starting with `LUNR_`.

The phpdoc should also mention how the parsing of the environment variables works. E.g. `PREFIX_FOO_BAR_BAZ` would be parsed into `$config['foo']['bar']['baz']`

Reply:
I don't think we should do the prefix. Things like DATABASE_HOST are specific enough. Prefixing it just makes it harder to work with.

Reply:
It has to be prefixed to wall it off against noise from the environment. `DATABASE_HOST` could be configuration for anything, it doesn't even have to be MySQL/MariaDB. The example we found for overriding environment variables from the environment, grafana, also uses `GF_<SECTION NAME>_<KEY>`, with `GF_` being the prefix.

I'm fine making the prefix optional, for cases where you are sure that no already defined variable is going to interfere with configuration, but I'd consider that the exception, not the rule.